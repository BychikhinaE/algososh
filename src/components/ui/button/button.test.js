import { Button } from "./button";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Тестирование компонента Button", () => {
  it("кнопка с текстом", () => {
    const buttonWithText = renderer.create(<Button text="text" />);
    expect(buttonWithText).toMatchSnapshot();
  });

  it("кнопка без текста", () => {
    const buttonWithoutText = renderer.create(<Button />);
    expect(buttonWithoutText).toMatchSnapshot();
  });

  it("заблокированная кнопка", () => {
    const buttonDisabled = renderer.create(<Button disabled={true} />);
    expect(buttonDisabled).toMatchSnapshot();
  });

  it("кнопка с индикацией загрузки", () => {
    const buttonWithLoader = renderer.create(<Button isLoader={true} />);
    expect(buttonWithLoader).toMatchSnapshot();
  });

  describe("корректность вызова колбека при клике на кнопку", () => {
    it("вызов колбека при клике на кнопку корректен", () => {
      const callBack = jest.fn();
      render(<Button onClick={callBack} />);
      const button = screen.getByTestId("button");
      fireEvent.click(button);
      expect(callBack).toHaveBeenCalled();
    });
  });
});
