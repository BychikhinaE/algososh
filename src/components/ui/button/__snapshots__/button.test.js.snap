// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
